core/store/src/trie/mem/updating.rs

```diff
@@ -38,7 +38,7 @@ pub enum UpdatedMemTrieNode {
     },
     /// Corresponds to either a Branch or BranchWithValue node.
     Branch {
-        children: [Option<OldOrUpdatedNodeId>; 16],
+        children: Box<[Option<OldOrUpdatedNodeId>; 16]>,
```

Comment:
In my understanding this code is sensitive to memory usage, but also the overhead of indirection shouldn't be that bad comparing to the size of the underlying array. Just in case tagging @robin-near to confirm.

Reply:
This part is not sensitive to memory usage. The parts sensitive to it are not easy to change like this, so don't worry about it.